YAXLib/YAXSerializerOfT.cs

```diff
@@ -139,7 +153,7 @@ public new T DeserializeFromFile(string fileName)
         /// <param name="obj">The object used as the base object in the next stage of de-serialization.</param>
         public void SetDeserializationBaseObject(T obj)
```

Comment:
Maybe we need object here (it was both there before)

Reply:
Yes, in particular because of
https://github.com/YAXLib/YAXLib/blob/5e7bcea6092afaa35af7b0b6a9059760bd060aab/YAXLib/YAXSerializer.cs#L1937
The generic overload does really make sense here, does it?

Reply:
Yeah I'm not sure, I never used SetDeserializationBaseObject

Reply:
Right, nor did I. Change this to `internal`?

Reply:
I think it's really needed. Will check the docs

Reply:
Docs: https://github.com/YAXLib/YAXLib/wiki#multi-stage-deserialization

Guess we need to check the tests 😅

Reply:
This is the only test. The assert is not really helping.

But it seems that it should be T and not object: https://github.com/YAXLib/YAXLib/blob/abdc4cb8866ead6061b0eecb523c008e5147bee8/YAXLibTests/DeserializationTest.cs#L351

Maybe I should test it with a different type. Maybe it gets a runtime error

Reply:
It should be T:

```
YAXLib.Exceptions.YAXObjectTypeMismatch : Expected an object of type 'MoreComplexBook2' but received an object of type 'MoreComplexBook2b'.
```